packages/horizontal-slider-js/src/horizontalSlider.tsx

```diff
@@ -0,0 +1,68 @@
+/** @jsx h */
+import {
+  createHorizontalSliderComponent,
+  generateHorizontalSliderId,
+  HorizontalSliderProps,
+  RecordWithObjectID,
+  updateNavigationButtonsProps,
+} from '@algolia/ui-components-horizontal-slider-vdom';
+import { createElement, Fragment, h, render } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+
+import { getHTMLElement } from './getHTMLElement';
+import { EnvironmentProps } from './types';
+
+const UncontrolledHorizontalSlider = createHorizontalSliderComponent({
+  createElement,
+  Fragment,
+});
+
+function HorizontalSlider<TObject extends RecordWithObjectID>(
+  props: HorizontalSliderProps<TObject>
+) {
+  const listRef = useRef<HTMLOListElement>(null);
+  const nextButtonRef = useRef<HTMLButtonElement>(null);
+  const previousButtonRef = useRef<HTMLButtonElement>(null);
+  const sliderIdRef = useRef(generateHorizontalSliderId());
+
+  useEffect(() => {
+    updateNavigationButtonsProps({
+      listRef,
+      nextButtonRef,
+      previousButtonRef,
+    });
+  });
+
+  return (
+    <UncontrolledHorizontalSlider
+      {...props}
+      listRef={listRef}
+      nextButtonRef={nextButtonRef}
+      previousButtonRef={previousButtonRef}
+      sliderIdRef={sliderIdRef}
+      updateNavigationButtonsProps={() =>
+        updateNavigationButtonsProps({
+          listRef,
+          nextButtonRef,
+          previousButtonRef,
+        })
+      }
+    />
+  );
+}
+
+export function horizontalSlider<TObject extends RecordWithObjectID>({
+  container,
+  environment = window,
+  ...props
+}: HorizontalSliderProps<TObject> & EnvironmentProps) {
+  const children = <HorizontalSlider {...props} />;
```

Comment:
should this be called children or component (it's only one element, maybe child)?

Reply:
We choosed `children` to keep it consistent with the return of our `recommendations` components

Reply:
`component` also makes sense to me. Both are fine IMO.